keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -887,19 +897,19 @@ routed according to the families assigned for services.
 
 1. For Headless with selector, the value of `spec.ipFamilies` will drive the
    endpoint selection. Endpoints will use the value of `ipFamiles[0]`.
-   EndpointSlices will use endpoints with all IP families of the Service.
-2. For Headless without selector the values of `spec.preferDualStack` and
-   `spec.ipFamilies` are defaulted and saved as normal, but are ignored by
-   kubernetes (since the standard endpoint controllers ignore such services).
-
-> In all cases, the values of `spec.preferDualStack`, `spec.ipFamilies`, and
-> `spec.clusterIPs` are validated and must be consistent.
+   EndpointSlices will use endpoints with all IP families of the Service. This
```

Comment:
I think we can loosen validation on headless, with or without selector, to allow `ipFamilies` to have 2 values.  They are not asking for a cluster IP, so they should not be restricted by which clusterIP CIDRs are configured.

This also makes headless+selectorless  consistent, and the only difference is the default.

And AFTER all this is in, we should consider whether the default for all headless should be "both families".   AFTER.

Reply:
*after* :-) I will leave this comment as unresolved as a reminder for our unfinished business.